include/fastdds/dds/subscriber/qos/DataReaderQos.hpp

```diff
@@ -105,8 +103,14 @@ class ReaderResourceLimitsQos
         std::swap(*this, reset);
     }
 
-    //!Matched publishers allocation limits.
-    fastrtps::ResourceLimitedContainerConfig matched_publisher_allocation;
+    //! Matched publishers allocation limits.
+    fastrtps::ResourceLimitedContainerConfig matched_publisher_allocation{ 1u };
+    //! SampleInfo allocation limits.
+    fastrtps::ResourceLimitedContainerConfig sample_infos_allocation{ 32u };
```

Comment:
On `matched_publisher_allocation`, I have incremented the default value for the initial allocations from 0 to 1. This will avoid a dynamic allocation when the first DataWriter is matched to the DataReader.

The 3 new parameters have similar default values to what other vendors have